### ISSUE TITLE
Add support for responding services

### ISSUE THREAD

**Issue by gagebenne** [esphome/feature-requests#2555]
**Describe the problem you have/What new integration you would like**
Home Assistant 2023.7 added the ability for a service to respond:
https://www.home-assistant.io/blog/2023/07/05/release-20237/#services-can-now-respond

esphome does not seem to support this new functionality:
https://esphome.io/components/api.html#user-defined-services

**Please describe your use case for this integration and alternatives you've tried:**
I have a pet feeder that is able to detect whether a feeding dispensed food or not. The feeding is a `esphome`-defined service, which I would like to respond whether the feeding was successful or not.

**Additional context**

**Comment by niahane** [comment]
Yes if there was a way to call a home assistant service and import its response into a response_variable: that would be fantastic.


globals:
  - id: forecast
    type: std::string
    restore_value: no
  
on..:
homeassistant.service:
service: weather.get_forecast
data:
  type: daily
target:
  entity_id: weather.home
on_response:
   then:
      - lambda: |-
             id(forecast) = x;
